tests/cpp_test/test_ChunkedArray.cpp

```diff
@@ -0,0 +1,262 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *
+ * Author: Alberto Ferreira
+ */
+#include <gtest/gtest.h>
+#include "../include/LightGBM/utils/ChunkedArray.hpp"
+
+using LightGBM::ChunkedArray;
+
+/*!
+  Helper util to compare two vectors.
+
+  Don't compare floating point vectors this way!
+*/
+template <typename T>
+testing::AssertionResult are_vectors_equal(const std::vector<T> &a, const std::vector<T> &b) {
+  if (a.size() != b.size()) {
+    return testing::AssertionFailure()
+      << "Vectors differ in size: "
+      << a.size() << " != " << b.size();
+  }
+
+  for (size_t i = 0; i < a.size(); ++i) {
+    if (a[i] != b[i]) {
+      return testing::AssertionFailure()
+        << "Vectors differ at least at position " << i << ": "
+        << a[i] << " != " << b[i];
+    }
+  }
+
+  return testing::AssertionSuccess();
+}
+
+
+class ChunkedArrayTest : public testing::Test {
+ protected:
```

Comment:
I'm intentionally not asking to follow Google coding style and apply `cpplint` to `tests/` folder with the aim not to delay the PR. Applying cpplint to `tests/` folder can be done in a separate PR without any harm to the development process.